### PR TITLE
Add ability to check and fix Django settings

### LINKED ISSUE #5: Local project checks before deploy
Currently we do check for the existence of a webapp type project before deploying.

However, this check should be much more comprehensive, and check for things that could create a broken deployment:
* Correct Django project structure and settings files.
* contents of settings file include minimum settings (static, media, database). Ideally this could be loaded in a python subprocess to fully interpret the variables, or the code could be parsed with regex/ast. It should be using our environment variables in prod and staging. 
* Correct wsgi file
* Requirements.txt

WordPress and static sites checking can be done separately, as those are much less delicate than Django projects. 

Finally, if errors are found, offer to fix them automatically if possible, or provide clear instructions on how to fix.

### PR DESCRIPTION
**Pull request by vsalvino**
Fixes #5 

* Add CodeRed Cloud Django settings templates.
* Check for wsgi.py. If it cannot be found, check if the `django_project` might be set incorrectly and offer to set it.
* Check for Django settings file. If it doesn't exist, offer to create it from a `settings.py`.
* Check and rewrite Django settings file(s) if they seem to be missing references to important environment variables such as `VIRTUAL_HOST` or `DB_HOST`.

Housekeeping:
* Move env to webapp `__init__` instead of args on utility methods.
* Upgrade pyinstaller
* Fix formatting of spec file (also include new templates dir)

Note: PATCH-ing the `custom_django_project` in coderedapi currently redeploys the webapp. In this case, that is undesired behavior as we are trying to fix this pre-deployment. We should probably change coderedapi before releasing this feature.

The code here seems quite inelegant and/or side-effect heavy. I guess that is the nature of what we are trying to do, with a lot of dependent nested logic. But open to suggestions on practical ways to organize it better (`api.py` file is getting a bit massive)

### CODE REVIEW

File: cr/api.py [hunk 1]
```diff
@@ -88,65 +91,273 @@ def __init__(self, handle: str, token: str):
...
+                    f"Your {self.app_type_name} project is missing a "
+                    f"`{project_file_relative}` file. "
+                    "Without this your app will not deploy correctly. "
+                    r"Continue anyways? [prompt.choices]\[y/N][/] ",
+                ).lower()
+            ):
+                raise UserCancelError()
+
+        # The following checks are only for Django-based projects.
+        if self.app_type not in [
```
Comment (jlchilders11) [inline cr/api.py]: Django app types is used in a few places, we should move that to some kind of global for ease of use.

**Comment by jlchilders11** [comment]
API.py is getting a bit dense and hard to follow. It seems the overwhelming majority of this size comes from django specific logic checks/fixes, so it would make sense logically to move these checks into a `django_checks.py` file.

**Comment by vsalvino** [comment]
I split out the individual checks into separate functions in utils. Then split high-level checking on the Webapp into separate functions. I'm a bit happier with it, and it should be mostly test-able now.

Also moved enums into `__init__.py` since they are basically constants.

File: cr/api.py (hunk 1 above)
Comment (vsalvino) [reply cr/api.py]: I refactored it a bit so the apptype checks are in one place.
